client/post-editor/editor-more-options/copy-post.jsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSite } from 'state/ui/selectors';
+import AccordionSection from 'components/accordion/section';
+import Button from 'components/button';
+import Dialog from 'components/dialog';
+import FormSectionHeading from 'components/forms/form-section-heading';
+import Gridicon from 'components/gridicon';
+import InfoPopover from 'components/info-popover';
+import PostSelector from 'my-sites/post-selector';
+
+class EditorMoreOptionsCopyPost extends Component {
+
+	static propTypes = {
+		site: PropTypes.object.isRequired,
+		translate: PropTypes.func,
+	};
+
+	constructor() {
+		super( ...arguments );
+		this.state = this.resetState();
+	}
+
+	resetState() {
+		return {
+			selectedPostId: null,
+			showDialog: false,
+		};
+	}
+
+	openDialog = event => {
+		event.preventDefault();
+
+		this.setState( {
+			showDialog: true,
+		} );
+	}
+
+	closeDialog = () => {
+		this.setState( this.resetState() );
+	}
+
+	setPostToCopy = post => {
+		this.setState( {
+			selectedPostId: post.ID,
+		} );
+	}
+
+	goToNewDraft = () => {
+		if ( '' !== this.state.selectedPostId ) {
+			page.redirect( `/post/${ this.props.site.slug }?copy=${ this.state.selectedPostId }` );
+			this.closeDialog();
+		}
+	}
+
+	render() {
+		const { translate, site } = this.props;
+		const buttons = [ {
+			action: 'cancel',
+			label: translate( 'Cancel' ),
+		}, {
+			action: 'copy',
+			label: translate( 'Overwrite' ),
+			isPrimary: true,
+			disabled: ! this.state.selectedPostId,
+			onClick: this.goToNewDraft,
+		} ];
+		return (
+			<AccordionSection className="editor-more-options__copy-post">
+				<div className="editor-drawer__label-text">
```

Comment:
Related to this, if the label were to wrap the `<Button />`, clicking the label would effect the same result as clicking the button directly. Possible usability/accessibility improvement?